src/sentry/models/project.py

```diff
@@ -313,6 +305,15 @@ def transfer_to(self, team):
                 team=team,
             )
 
+        # We only need to delete ReleaseProjects when moving to a different
```

Comment:
I should update this comment.

Also note, I moved this to _after_ the actual transfer since there's technically a race condition where the rows are deleted first, then new ones could pop back in, before the project is actually moved.

This just moves it to after so the worst case is we overly eagerly delete an extra row instead of leaving 1 in a bad state.